sympy/sets/fancysets.py

```diff
@@ -446,20 +456,16 @@ def __new__(cls, *args):
             raise ValueError("step cannot be 0")
 
         start, stop, step = slc.start or 0, slc.stop, slc.step or 1
-        try:
-            start, stop, step = [
-                w if w in [S.NegativeInfinity, S.Infinity]
-                else sympify(as_int(w))
-                for w in (start, stop, step)]
-        except ValueError:
-            raise ValueError(filldedent('''
-    Finite arguments to Range must be integers; `imageset` can define
-    other cases, e.g. use `imageset(i, i/10, Range(3))` to give
-    [0, 1/10, 1/5].'''))
 
-        if not step.is_Integer:
+        start, stop, step = map(_sympify, [start, stop, step])
+
+        tmp_obj = Basic(start, stop, step)
+        if tmp_obj.atoms(Float) and (not tmp_obj.atoms(Symbol)):
+            raise ValueError('Floats are not permitted in Range.')
```

Comment:
I would use assumptions on `start`, `stop`, `step`. If they have `.is_integer == False`, raise the exception. If they have `.is_integer == None`, don't, as they may or may not be integers.

Reply:
@smichr Please provide your views, what should I do? I am supportive of @Upabjojr view for now. Let me know of what you think?